pulp_rpm/app/depsolving.py

```diff
@@ -835,7 +836,7 @@ def resolve_dependencies(self, unit_repo_map):
                 passthrough[repo].add(unit.pk)
 
         self._pool.createwhatprovides()
-        flags = solv.Job.SOLVER_INSTALL | solv.Job.SOLVER_SOLVABLE
```

Comment:
Let's add some comments for the multiversion flag to the block on lines 812-817.

Reply:
And we can delete the commented-out code which used to handle this, `_handle_same_name` and the conditional that called into it.

Reply:
[This](https://github.com/pulp/pulp_rpm/pull/2080/files#diff-d200f526f2c6666a66ddb034499ce27c877c8fd607ccb03bd0719733b1dac68dR780) one *might* be covered by the other flag you've added, I'm not sure.

Reply:
Good idea RE adding discussion to the overview-comment. Right now the prob that
this PR is having, is I added MULTIVERSION **in the wrong place** . Once I fix it everything is happier.

RE commented-out code - there's a lot right now, since we decided to push off handle_problems() to later/never. I can remove as part of this PR - or, leave it alone for a cleanup PR "later"?

Reply:
I'm fine with leaving the commented code, no strong feelings.

re: right / wrong place, I'm having a difficult time understanding the DNF code but it seems like this is either provided at job construction time just like it is here (before)? 

https://github.com/rpm-software-management/libdnf/blob/9a0e17562b19586b3ffa70fa93eb961b558794c7/libdnf/goal/Goal.cpp

Reply:
Setting it per-job, instead of at the pool-level, failed (pretty hard) for me. Like you, I don't have my head wrapped around exactly **why** - but that's what observation/experimentation is telling me.

Reply:
solv.Job.SOLVER_MULTIVERSION, when set per-job, tells libsolv to **skip doing dependency resolution** for the entity - which is...bad. Set on the Pool, it does seem to help us avoid a raft of false-positive warnings on depsolv.

Reply:
Maybe SOLVER_TRANSACTION_MULTIINSTALL is the one I'm thinking of?